lib/WebpackOptionsDefaulter.js

```diff
@@ -166,25 +170,16 @@ class WebpackOptionsDefaulter extends OptionsDefaulter {
 		this.set("node.__filename", "mock");
 		this.set("node.__dirname", "mock");
 
-		this.set(
-			"performance",
-			"make",
-			options => (isProductionLikeMode(options) ? false : undefined)
-		);
-		this.set("performance", "call", value => {
-			if (typeof value === "boolean") {
-				return value;
+		this.set("performance", "make", options => {
+			if (isWebLikeTarget(options) && isProductionLikeMode(options)) {
+				return {};
 			} else {
-				return Object.assign({}, value);
+				return false;
 			}
 		});
 		this.set("performance.maxAssetSize", 250000);
 		this.set("performance.maxEntrypointSize", 250000);
-		this.set(
-			"performance.hints",
-			"make",
-			options => (isProductionLikeMode(options) ? "warning" : false)
-		);
```

Comment:
For a configuration like

``` js
performance: {
  maxAssetSize: 100000
}
```

warnings should only be triggered in production mode.